src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -224,42 +245,147 @@ public void DirectoryInfo_Create_NullDirectorySecurity()
-        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.ReadData, AccessControlType.Allow)]
-        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.Write, AccessControlType.Allow)]
-        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.Write, AccessControlType.Deny)]
         [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.FullControl, AccessControlType.Deny)]
         public void DirectoryInfo_Create_DirectorySecurityWithSpecificAccessRule(
             WellKnownSidType sid,
             FileSystemRights rights,
             AccessControlType controlType)
         {
-
             DirectorySecurity security = GetDirectorySecurity(sid, rights, controlType);
             VerifyDirectorySecurity(security);
         }
 
         #endregion
 
+        #region FileInfo Create
+
+        [Fact]
+        public void FileInfo_Create_NullFileInfo()
+        {
+            FileInfo info = null;
+            FileSecurity security = new FileSecurity();
+
+            if (PlatformDetection.IsFullFramework)
+            {
+                AssertExtensions.Throws<ArgumentNullException>(() =>
+                    FileSystemAclExtensions.Create(info, FileMode.Create, FileSystemRights.WriteData, FileShare.Read, DefaultBufferSize, FileOptions.None, security),
+                    "Value cannot be null.\r\nParameter name: fileInfo");
+            }
+            else
+            {
+                AssertExtensions.Throws<ArgumentNullException>(() =>
+                    info.Create(FileMode.Create, FileSystemRights.WriteData, FileShare.Read, DefaultBufferSize, FileOptions.None, security),
+                    "Value cannot be null. (Parameter 'fileInfo')");
+            }
+        }
+
+        [Fact]
+        public void FileInfo_Create_DefaultFileSecurity()
+        {
+            FileSecurity security = new FileSecurity();
+            VerifyFileSecurity(security);
+        }
+
+        [Fact]
+        public void FileInfo_Create_NullFileSecurity()
+        {
+            FileInfo info = new FileInfo("path");
+            if (PlatformDetection.IsFullFramework)
+            {
+                AssertExtensions.Throws<ArgumentNullException>(() =>
+                    FileSystemAclExtensions.Create(info, FileMode.Create, FileSystemRights.WriteData, FileShare.Read, DefaultBufferSize, FileOptions.None, null),
+                    "Value cannot be null.\r\nParameter name: fileSecurity");
+            }
+            else
+            {
+                AssertExtensions.Throws<ArgumentNullException>(() =>
+                    info.Create(FileMode.Create, FileSystemRights.WriteData, FileShare.Read, DefaultBufferSize, FileOptions.None, null),
+                    "Value cannot be null. (Parameter 'fileSecurity')");
+            }
+        }
+
+        [Fact]
+        public void FileInfo_Create_NotFound()
+        {
+            FileInfo info = new FileInfo(@"W:\I\Do\Not\Exist\file.txt");
```

Comment:
Same comment as above with the file path in temp.

Reply:
Done.